conbench/entities/run.py

```diff
@@ -249,6 +247,30 @@ def get_default_baseline_run(
 
         return Run.get(closest_run_id)
 
+    def get_candidate_baseline_runs(self) -> Dict[str, Optional[_FoundBaselineRun]]:
+        """Return information about a few different candidate baseline runs, including
+        on the parent commit, fork-point commit, and head-of-default-branch commit.
+
+        See docstring of _get_baseline_run_from_baseline_commit() for how these are
+        found.
+        """
+        return {
+            "parent": self._get_baseline_run_from_baseline_commit(
+                self.commit.get_parent_commit()
+            ),
+            "fork_point": self._get_baseline_run_from_baseline_commit(
+                self.commit.get_fork_point_commit()
```

Comment:
Yes!

Reply:
If the contender is on the default branch, this will return itself, right? I think we talked about returning null instead; do we want to do that here? Or pop off all invalid/silly baselines (e.g. head of default (or whatever we call it) to itself)?

Reply:
Yes, `fork_point` should be null when the contender run is on the default branch.

Not sure about `head_of_default` when the contender run is on the head of default. Yeah, I suppose we null out any where the contender ID and baseline ID are the same.

Reply:
> `fork_point` should be null when the contender run is on the default branch.

I super strongly agree with this. I know it's possible to get in the git API, but it's definitely confusing + will lead to people being confused what we mean and what it's useful for.

For `head_of_default` IMO we should return whatever is the head of default (that has benchmark results) regardless. Even if it's the same ID, that's a decent signal that the thing you're looking at is the latest-and-greatest. It's also totally fine that this value will change over time since it is truly intended to track `HEAD`. So sometimes the run id might be the same for `self` and `head_of_default` but 1 hour later, `head_of_default` could be different.

Reply:
>For head_of_default IMO we should return whatever is the head of default (that has benchmark results) regardless.

Something that came up while I was working on this: do we still want to return a `head_of_default` baseline for contender runs that are on the default branch already? Like on this picture, if you're asking for candidate baseline runs of contender run 4:
<img width="805" alt="image" src="https://user-images.githubusercontent.com/16600275/233732428-1e50f3df-6333-480b-bb61-18fa35f6da2b.png">
Obviously, for a contender on 4, then `parent` baseline is 3 and `fork_point` baseline is `null`. But do we really want to say the `head_of_default` baseline of contender run 4 is actually 5? If someone ends up hitting `/api/compare/5...4` for that comparison analysis, then when calculating the z-scores of 4, 4's data will be used as part of the baseline distribution when calculating z-scores (as well as the contender data point), which seems like a smell to me. Instead, I was thinking we should have the `head_of_default` baseline be `null` for default-branch commits too.

Not super tied to that decision.

>I think [nulling out any where the contender ID and baseline ID are the same] is trying to find a rule when we don't need one.

Yeah, I changed the code so this isn't the case. The current code is more aligned to what you're thinking, I think.

Reply:
Changed this in https://github.com/conbench/conbench/pull/1275.